api/src/main/java/org/ehcache/spi/serialization/StatefulSerializer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.spi.serialization;
+
+import org.ehcache.spi.persistence.StateRepository;
+
+/**
+ * Implementations of this interface can have their state maintained in a {@code StateRepository}.
+ * The state will be maintained by the authoritative tier of the cache for which this is configured.
+ * When used within the default serialization provider, there is an additional constructor requirement.
+ * The implementations must define a constructor that takes in a {@code ClassLoader}.
+ * Post instantiation, the state repository will be injected with the {@code init} method invocation.
```

Comment:
I would add that this call will happen before any interaction with the `Serializer` interface.